src/set_module.f90

```diff
@@ -514,6 +512,8 @@ subroutine write_set_constrain(ictrl)
          write(ictrl,'(3x,"dihedral:",1x,4(i0,",",1x),g0,1x,"# in Degree")') &
             atconstr(1,i),atconstr(2,i),atconstr(3,i),atconstr(4,i), &
             valconstr(i)*180.0_wp/pi
+      case default ! this is bad, we don't want this to happen
+         call raise('E','This is an internal error, please report this!')
```

Comment:
Error handling should happen at the top of the select case, for readability reasons. Could you please revert the changes in this source file?